dali/base/dameta.cpp

```diff
@@ -22,13 +22,19 @@
 
 
 //More to a more central location
-IPropertyTree * queryHostGroup(const char * name)
+IPropertyTree * queryHostGroup(const char * name, bool required)
 {
-    if (isEmptyString(name))
-        return nullptr;
-    VStringBuffer xpath("storage/hostGroups[@name='%s']", name);
-    IPropertyTree & global = queryGlobalConfig();
-    return global.queryPropTree(xpath);
+    if (!isEmptyString(name))
+    {
+        VStringBuffer xpath("storage/hostGroups[@name='%s']", name);
+        IPropertyTree & global = queryGlobalConfig();
+        IPropertyTree * match = global.queryPropTree(xpath);
+        if (match)
+            return match;
+    }
+    if (required)
+        throw makeStringExceptionV(0, "No entry found for hostGroup: '%s'", name ? name : "<null>");
```

Comment:
minor: are we tending to default to error code 0, or -1?

Reply:
We should be defining error codes for all the errors, rather than being lazy!

Reply:
I would expect this validation to take place (primarily) in the helm templates.  It would be nice for invalid configuration to cause aborts early rather than starting up and then failing.

Reply:
>We should be defining error codes for all the errors, rather than being lazy!

agree, but even when being lazy - better to use a consistent default ;)
otherwise might think 0 has a special meaning

Reply:
changed